src/test/java/reactor/kafka/AbstractKafkaTest.java

```diff
@@ -194,33 +209,73 @@ protected void waitForTopic(String topic, int partitions, boolean resetMessages)
         }
     }
 
+    private void waitForTopic(String topic) {
+        Map<String, Object> props = new HashMap<>();
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers());
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class);
+        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class);
+        props.put(ProducerConfig.MAX_BLOCK_MS_CONFIG, 1000);
+        try (KafkaProducer<byte[], byte[]> producer = new KafkaProducer<>(props)) {
+            int maxRetries = 10;
+            boolean done = false;
+            for (int i = 0; i < maxRetries && !done; i++) {
+                List<PartitionInfo> partitionInfo = producer.partitionsFor(topic);
+                done = !partitionInfo.isEmpty();
+                for (PartitionInfo info : partitionInfo) {
+                    if (info.leader() == null || info.leader().id() < 0)
+                        done = false;
+                }
+            }
+            assertTrue("Timed out waiting for topic", done);
+        }
+    }
+
     protected void waitForBrokers() {
-        embeddedKafka.waitForBrokers();
+        int maxRetries = 50;
+        for (int i = 0; i < maxRetries; i++) {
+            try {
+                bootstrapServers();
+                break;
+            } catch (Exception e) {
+                reactor.kafka.util.TestUtils.sleep(500);
+            }
+        }
     }
 
-    protected void shutdownKafkaBroker() {
-        embeddedKafka.shutdownBroker(brokerId);
+    protected void assumeBrokerRestartSupport() {
+        Assume.assumeTrue("supports broker restart", false);
     }
 
-    protected void startKafkaBroker() {
-        embeddedKafka.startBroker(brokerId);
+    protected void shutdownKafkaBroker() {
+        assumeBrokerRestartSupport();
+        // TODO embeddedKafka.shutdownBroker(brokerId);
```

Comment:
not sure this TODO is still needed

Reply:
Right. Replaced with `Assert.fail` so that (if we ever decide to add the broker restart scenario) it will remind us that we need to implement these methods 👍